exercises/diamond/test/Tests.hs

```diff
@@ -3,15 +3,21 @@
 import Data.Foldable     (for_)
 import Test.Hspec        (Spec, describe, it, shouldBe)
 import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Test.QuickCheck   (Gen, oneof, forAll)
+import Data.List         ((\\))
+import Data.Maybe        (isNothing)
 import Diamond (diamond)
 
 main :: IO ()
 main = hspecWith defaultConfig {configFastFail = True} specs
 
 specs :: Spec
-specs = describe "diamond" $ for_ cases test
+specs = describe "diamond" $ do
+  it "non-Alpha characters should produce `Nothing`" $
+    forAll genNonAlphaChars $
+      isNothing . diamond
+  for_ cases test
```

Comment:
In #735 we move the property-based tests last, but considering [Mark Seemann's blog post](https://blog.ploeh.dk/2015/01/10/diamond-kata-with-fscheck/) on exactly this:

> The no-brain approach to Example-Driven Development is to start with 'A', then 'B', and so on. Exactly as Seb Rose predicts, when you approach the problem like this, when you reach 'C', it no longer seems reasonable to hard-code the responses, but then the entire complexity of the problem hits you all at once. It's quite hard to do incremental development by going through the 'A', 'B', 'C' progression.

So I think their position first are fine, as long as the generators are listed last.

```suggestion

    for_ cases test
```